src/service/db.py

```diff
@@ -13,26 +13,44 @@ class SongServiceByDB(SongService):
     def __init__(
         self,
         googleCredentials: Mapping[str, str],
-        defaultSpreadsheetId: str = "",
+        spreadsheetIds: Mapping[str, str],
+        defaultGroup: str = "",
         cache: Optional[Cache] = None,
     ) -> None:
-        self.defaultSpreadsheetId = defaultSpreadsheetId
+        self.spreadsheetIds = spreadsheetIds
+        self.defaultSpreadsheetId = spreadsheetIds[defaultGroup]
         self.service = SongDB(googleCredentials, cache=cache)
         self.cache = cache
 
+        self._create_song_mappings()
+
+    def _create_song_mappings(self):
+        self.songMappings = {
+            self._to_song_key(song): {"group": group, "name": song}
+            for group, id in self.spreadsheetIds.items()
+            for song in self.service.list_song_names(id)
+        }
+
     @with_cache("SongServiceByDB::get_song")
-    def get_song(self, songName: str, spreadsheetId: str = "") -> Song:
-        if spreadsheetId == "":
-            spreadsheetId = self.defaultSpreadsheetId
+    def get_song(self, songName: str, spreadsheet: str = "") -> Song:
+        spreadsheetId = self.spreadsheetIds.get(spreadsheet, self.defaultSpreadsheetId)
 
         songKeyToFind = self._to_song_key(songName)
 
-        for existingSongName in self.service.list_song_names(spreadsheetId):
-            if self._to_song_key(existingSongName) == songKeyToFind:
-                return self.service.get_song(
-                    self.defaultSpreadsheetId if not spreadsheetId else spreadsheetId,
-                    existingSongName,
-                )
+        if songKeyToFind in self.songMappings:
+            group = self.songMappings[songKeyToFind]["group"]
+            existingSongName = self.songMappings[songKeyToFind]["name"]
+            spreadsheetId = self.spreadsheetIds[group]
+            return self.service.get_song(spreadsheetId, existingSongName)
+        else:
+            for existingSongName in self.service.list_song_names(spreadsheetId):
+                if self._to_song_key(existingSongName) == songKeyToFind:
+                    return self.service.get_song(
+                        self.defaultSpreadsheetId
+                        if not spreadsheetId
+                        else spreadsheetId,
```

Comment:
Seems redundant since we've already made sure `spreadsheetId` is populated above

Reply:
I think this whole section is probably redundant, removed.